File: WarehouseManager/src/main/java/hu/elte/WarehouseManager/model/Request.java
package hu.elte.WarehouseManager.model;

import lombok.Data;

import javax.persistence.*;

@Entity
@Data
@Table(name = "REQUESTS")
public class Request {

    @Id
    @Column(name = "REQUEST_ID")
    private Integer requestID;

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "USER_ID")
    private User owner;

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "PRODUCT_NAME")
    private Product product;

    @Column(name = "ORDERED_AMOUNT")
    private Integer orderedAmount;
}

File: WarehouseManager/warehouse-manager-client/src/app/model/request.ts
import { User } from '../model/user';
import { Product } from '../model/product';

export class Request {

    // fields
    requestID: number;
    owner: User;
    product: Product;
    orderedAmount: number;

    // constructor
    constructor(
        requestID:number,
        owner:User,
        product:Product,
        orderedAmount:number) { 

            this.requestID = requestID; 
            this.owner = owner;
            this.product = product;
            this.orderedAmount = orderedAmount;
     }
}